pkg/middleware/auth.go

```diff
@@ -90,6 +101,95 @@ func (c JWTVerifier) CheckToken(token string) (*jwt.Token, error) {
 	return tk, nil
 }
 
+// JWTVerifierURL checks token with JWT against JWKS, implements Verifier
+type JWTVerifierURL struct {
+	jwksURL string
+
+	parser   *jwt.Parser
+	jwksKeys *jwk.Set
+	jwksLock sync.RWMutex
+}
+
+func (c *JWTVerifierURL) startTicker(ticker time.Duration) {
+
+	for range time.Tick(ticker) {
+		err := c.updateKeySet()
+		if err != nil {
+			log.Warn().Msg("unable to update JWKS key set")
+		}
+	}
+}
+
+func (c *JWTVerifierURL) updateKeySet() error {
+
+	c.jwksLock.Lock()
+	defer c.jwksLock.Unlock()
+
+	set, err := jwk.Fetch(c.jwksURL)
+	if err != nil {
+		return err
+	}
+	c.jwksKeys = set
+	return nil
+}
+
+func (c *JWTVerifierURL) getKeySet() *jwk.Set {
+	c.jwksLock.RLock()
+	defer c.jwksLock.RUnlock()
+	return c.jwksKeys
+
+}
+
+// NewJWTVerifierURL creates JWTVerifierURL with JWKS URL
+func NewJWTVerifierURL(jwksURL string, updateInterval time.Duration) *JWTVerifierURL {
+
+	http.DefaultClient = &http.Client{Timeout: 10 * time.Second}
```

Comment:
Is this going to affect the timeout or other settings on other clients?

Reply:
I don't know if it's worthwhile to introduce another parameter for this timeout.